mesonbuild/dependencies/base.py

```diff
@@ -1723,8 +1723,9 @@ def __init__(self, name: str, command: typing.Optional[typing.List[str]] = None,
 
         if not silent:
             if self.found():
-                mlog.log('Program', mlog.bold(name), 'found:', mlog.green('YES'),
-                         '(%s)' % ' '.join(self.command))
+                if not self.path.startswith(search_dir):
```

Comment:
That will fail if search_dir is None, and that's the default value, so I gues that can happen. Also no need of nested if here.

Reply:
right, it needs to handle None. Nested if is necessary though else it will log it didn't find it.